EasyPost.Tests/CarrierAccountTest.cs

```diff
@@ -1,131 +1,133 @@
+            }
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Read]
         public async Task TestRetrieve()
         {
-            _vcr.SetUpTest("retrieve");
+            UseVCR("retrieve");
 
             CarrierAccount carrierAccount = await CreateBasicCarrierAccount();
 
-            CarrierAccount retrievedCarrierAccount = await CarrierAccount.Retrieve(carrierAccount.id);
+            CarrierAccount retrievedCarrierAccount = await Client.CarrierAccount.Retrieve(carrierAccount.id);
 
-            Assert.IsInstanceOfType(retrievedCarrierAccount, typeof(CarrierAccount));
-            Assert.AreEqual(carrierAccount, retrievedCarrierAccount);
+            Assert.IsType<CarrierAccount>(retrievedCarrierAccount);
+            Assert.Equal(carrierAccount, retrievedCarrierAccount);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Read]
         public async Task TestTypes()
         {
-            _vcr.SetUpTest("types");
+            UseVCR("types");
 
-            List<CarrierType> types = await CarrierType.All();
+            List<CarrierType> types = await Client.CarrierType.All();
 
-            foreach (var item in types)
+            foreach (CarrierType item in types)
             {
-                Assert.IsInstanceOfType(item, typeof(CarrierType));
+                Assert.IsType<CarrierType>(item);
             }
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestUpdate()
         {
-            _vcr.SetUpTest("update");
+            UseVCR("update");
+
 
             CarrierAccount carrierAccount = await CreateBasicCarrierAccount();
 
             string testDescription = "my custom description";
 
-            Dictionary<string, object> carrierAccountData = new Dictionary<string, object>
+            Dictionary<string, object> carrierAccountData = new Dictionary<string, object>()
             {
                 {
-                    "description", testDescription
+                    "description",
+                    testDescription
                 }
             };
-            await carrierAccount.Update(carrierAccountData);
+            carrierAccount = await carrierAccount.Update(carrierAccountData);
+
+            Assert.IsType<CarrierAccount>(carrierAccount);
+            Assert.StartsWith("ca_", carrierAccount.id);
+            Assert.Equal(testDescription, carrierAccount.description);
+        }
+
+        [Fact]
+        [CrudOperations.Delete]
+        public async Task TestDelete()
+        {
+            UseVCR("delete");
 
-            Assert.IsInstanceOfType(carrierAccount, typeof(CarrierAccount));
-            Assert.IsTrue(carrierAccount.id.StartsWith("ca_"));
-            Assert.AreEqual(testDescription, carrierAccount.description);
+            CarrierAccount carrierAccount = await CreateBasicCarrierAccount();
+
+            await carrierAccount.Delete();
+
+            // TODO: Assert something
```

Comment:
Issue: Let's assert something now. Whatever this returns at present, and we can update the assertions later when we fix deleting. A test with no assertion is not a test.

Reply:
There's nothing to assert. `Delete` returns void. The unit test is inherently asserting that no error was thrown on the API call by not failing.

Reply:
Does C# not have an "assert no error" like other libs? If so, we should use that here to signify that that is our explicit test. I'd like to avoid implicit tests where possible.

Reply:
There's some interesting discussion about that in [this thread](https://github.com/xunit/xunit/issues/188). I have an idea about how to explicitly check for no exception.

Reply:
Resolved.